### PR TITLE
Chore: Add documentation for context

### PR DESCRIPTION
**Pull request by sangee2004**
Add documentation for context

### CODE REVIEW

**Review by g-linville — APPROVED** [review]

File: docs/docs/03-tools/05-context.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+Here is an example of a context provider tool that uses args to decide which search tool to use when answering the user provided queries:
+
+```yaml
+# context_with_arg.gpt
+name: search_context
+export: github.com/gptscript-ai/search/duckduckgo, github.com/gptscript-ai/search/brave, sys.http.html2text?
+args: search_tool: tool to search with
+
+#!/bin/bash
+echo You are an expert web researcher with access to the ${search_tool} Search tool.If the search tool fails to return any information stop execution of the script with message "Sorry! Search did not retrun any results" . Feel free to get the contents of the returned URLs in order to get more information. Provide as much detail as you can. Also return the source of the search results.
```
Comment (tylerslaton) [inline docs/docs/03-tools/05-context.md:50]:
```suggestion
echo You are an expert web researcher with access to the ${search_tool} Search tool. If the search tool fails to return any information stop execution of the script with message "Sorry! Search did not return any results" . Feel free to get the contents of the returned URLs in order to get more information. Provide as much detail as you can. Also return the source of the search results.
```

File: docs/docs/03-tools/05-context.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+Here is a simple example of a context provider tool that provides additional context to search tool:
+
+```yaml
+# my-search-context-tool.gpt
+name: search_context
+export: sys.http.html2text?
+
+#!/bin/bash
+echo You are an expert web researcher with access to the Search tool.If the search tool fails to return any information stop execution of the script with message "Sorry! Search did not retrun any results" . Feel free to get the contents of the returned URLs in order to get more information. Provide as much detail as you can. Also return the source of the search results.
```
Comment (tylerslaton) [inline docs/docs/03-tools/05-context.md:20]:
```suggestion
echo You are an expert web researcher with access to the Search tool. If the search tool fails to return any information stop execution of the script with message "Sorry! Search did not return any results". Feel free to get the contents of the returned URLs in order to get more information. Provide as much detail as you can. Also return the source of the search results.
```

Comment (sangee2004) [reply docs/docs/03-tools/05-context.md]: I have made the suggested changes.

**Review by tylerslaton — APPROVED** [review]

**Review by njhale — APPROVED** [review]

File: docs/docs/03-tools/05-context.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+GPTScript supports context provider tools which can be used to provide additional context for tool execution. Output from the context tool gets prepended to the instruction of the current tool that uses the context tool.
+
+## Writing a Context Provider Tool
+
+Here is a simple example of a context provider tool that provides additional context to search tool:
+
+```yaml
+# my-search-context-tool.gpt
+name: search_context
```
Comment (ibuildthecloud) [inline docs/docs/03-tools/05-context.md:11]: Can we not put `name:` as it's not required.  I more care about below not using the `from` sytnax.  So the example below should just be `context: ./my-search-context-tool.gpt`

File: docs/docs/03-tools/05-context.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+echo You are an expert web researcher with access to the Search tool.If the search tool fails to return any information stop execution of the script with message "Sorry! Search did not return any results". Feel free to get the contents of the returned URLs in order to get more information. Provide as much detail as you can. Also return the source of the search results.
+
+```
+
+## Using a Context Provider Tool
+
+Continuing with the above example, this is how you can use it in a script:
+
+```yaml
+context: search_context from  my-search-context-tool.gpt
```
Comment (ibuildthecloud) [inline docs/docs/03-tools/05-context.md:24]: change to `context: ./my-search-context-tool.gpt`

File: docs/docs/03-tools/05-context.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
+# Context
+
+GPTScript supports context provider tools which can be used to provide additional context for tool execution. Output from the context tool gets prepended to the instruction of the current tool that uses the context tool.
```
Comment (ibuildthecloud) [inline docs/docs/03-tools/05-context.md:3]: Can we document that exported tools from a context tools are added to the current tool's tools. (wow that was a bad way to say it).  Also can we document that you can just do `context: foo.txt` if you just want to add a string. This is a simple way to share prompts across tools.

Comment (cloudnautique) [inline docs/docs/03-tools/05-context.md:3]: We shouldn't use "context provider tools" in the definition as those are not yet defined.

I think we are trying to convey that context:
- Provides a mechanism to use prompt information across many tools as an import mechanism.
- Provide additional information to the calling tool on when to use a specific tool. (Helpful for using new CLIs for example)
- These can be static txt or dynamic gptscripts.

File: docs/docs/03-tools/05-context.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+## Writing a Context Provider Tool
+
+Here is a simple example of a context provider tool that provides additional context to search tool:
```
Comment (cloudnautique) [inline docs/docs/03-tools/05-context.md:7]: Note that they are just regular GPTScript tools, and any valid gptscript field can be used.

**Comment by cloudnautique** [comment]
I would also add a txt example to the docs. 

Don't actually use this, but it could look like:
pirate.txt
```text
talk like a pirate
```

Tool1:
```
Context: pirate.txt

Tell a joke
```

Tool2:
```
Context: pirate.txt

Say Hello
```

Showing you can reduce boiler plate and change behavior quickly

File: docs/docs/03-tools/05-context.md (hunk 3 above)
Comment (sangee2004) [reply docs/docs/03-tools/05-context.md]: Made the suggested change.

File: docs/docs/03-tools/05-context.md (hunk 4 above)
Comment (sangee2004) [reply docs/docs/03-tools/05-context.md]: Made the suggested change.

File: docs/docs/03-tools/05-context.md (hunk 5 above)
Comment (sangee2004) [reply docs/docs/03-tools/05-context.md]: Following documentation is now added:
```
GPTScript provides a mechanism to share prompt information across many tools using the tool parameter `context`. It is used to provide additional information to the calling tool on when to use a specific tool by prepending the `context` to the instruction of the calling tool.

- Context can point to a static text or a GPTScript.
- Context tools are just regular GPTScript tools, and any valid gptscript field can be used.
- Exported tools from a context tools are available as tools to the calling tool.
- When context points to a GPTScript tool, output from the context tool gets prepended to the instruction of the calling tool.
```

Comment (sangee2004) [reply docs/docs/03-tools/05-context.md]: Following documentation has been added now:'
```
GPTScript provides a mechanism to share prompt information across many tools using the tool parameter `context`. It is used to provide additional information to the calling tool on when to use a specific tool by prepending the `context` to the instruction of the calling tool.

- Context can point to a static text or a GPTScript.
- Context tools are just regular GPTScript tools, and any valid gptscript field can be used.
- Exported tools from a context tools are available as tools to the calling tool.
- When context points to a GPTScript tool, output from the context tool gets prepended to the instruction of the calling tool.
```

File: docs/docs/03-tools/05-context.md (hunk 6 above)
Comment (sangee2004) [reply docs/docs/03-tools/05-context.md:44]: This information is now included.

**Comment by sangee2004** [comment]
@cloudnautique I have included an example for multiple tools using txt file as context.

**Review by cloudnautique — APPROVED** [review]

**Pull request merged**